src/go/wsl-helper/pkg/wsl-utils/utf16writer_windows.go

```diff
@@ -0,0 +1,26 @@
+package wslutils
+
+import (
+	"io"
+	"unsafe"
+
+	"golang.org/x/sys/windows"
+)
+
+// utf16Writer is a writer that attempts to convert writes in UTF-16 to UTF-8.
+type utf16Writer struct {
+	io.Writer
+}
+
+func (w *utf16Writer) Write(p []byte) (int, error) {
+	output := windows.UTF16PtrToString(
+		(*uint16)(unsafe.Pointer(unsafe.SliceData(append(p, 0, 0)))),
+	)
+	n, err := w.Writer.Write(
+		unsafe.Slice(unsafe.StringData(output), len(output)),
+	)
+	if err != nil {
```

Comment:
These 4 lines can be simplifed to `return n * w, err`

Reply:
Fixed. (Due to issues with wrapping nils of different types, I defaulted to doing the extra check.)